src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -812,7 +813,26 @@ private static class ImageHelper implements ImageProvider {
                     ByteOrder order,
                     Path packagedModulesPath,
                     boolean ignoreSigning) throws IOException {
-            this.order = order;
+            if (order != null) {
+                this.order = order;
+            } else {
+                // Use the java.base module of the target platform to determine the endianness
+                // of the target image
+                String targetPlatform = null;
+                Optional<ResolvedModule> javaBase = cf.findModule("java.base");
+                if (javaBase.isPresent()) {
+                    ModuleReference ref = javaBase.get().reference();
+                    if (ref instanceof ModuleReferenceImpl modRefImpl
+                            && modRefImpl.moduleTarget() != null) {
+                        targetPlatform = modRefImpl.moduleTarget().targetPlatform();
```

Comment:
Hello Alan,

>  If getDefaultModulePath() is the parent of the location containing java.base then it can use the current endianness.

I've now updated this PR to implement this suggestion. The `ModuleTarget` attribute in java.base's `module-info.class` will only be checked if the java.base isn't from the current execution environment's default jmods location.

Reply:
> Also just to say that if it would be an error if java.base is not in the Configuration, same thing if there is no packaged module for java.base.

I've now added `assert`s to expect this module to be present.